app/src/main/java/com/kickstarter/viewmodels/SearchViewModel.java

```diff
@@ -96,6 +107,30 @@ public ViewModel(final @NonNull Environment environment) {
       final Observable<String> query = params
         .map(DiscoveryParams::term);
 
+      final Observable<List<Project>> projects = Observable.merge(
+        this.popularProjects.asObservable(),
+        this.searchProjects.asObservable()
+      );
+
+      Observable.combineLatest(search, projects, Pair::create)
+        .compose(Transformers.takePairWhen(this.tappedProject))
+        .map(searchTermAndProjectsAndTappedProject -> {
+          final String searchTerm = searchTermAndProjectsAndTappedProject.first.first;
+          final List<Project> currentProjects = searchTermAndProjectsAndTappedProject.first.second;
+          final Project tappedProject = searchTermAndProjectsAndTappedProject.second;
+
+          if (searchTerm.length() == 0) {
+            return tappedProject == currentProjects.get(0)
+              ? Pair.create(tappedProject, RefTag.searchPopularFeatured())
+              : Pair.create(tappedProject, RefTag.searchPopular());
+          } else {
+            return tappedProject == currentProjects.get(0)
+              ? Pair.create(tappedProject, RefTag.searchFeatured())
+              : Pair.create(tappedProject, RefTag.search());
+          }
+        })
```

Comment:
also, `.compose(bindToLifecycle())` should be the last operator chained on before `.subscribe()`